### PR TITLE
feat(ApiAuth): reduce probability of token expiry moments after retrieval

### PR DESCRIPTION
**Pull request by richardkeit**
There is currently no buffer on the time after getting tokens.
That being said, if you complete the `.requests_auth` flow and the token expired 1 second after that - you're running around with an expired set of creds being none the wiser.

Example out in the wild:
image

### CODE REVIEW

File: staxapp/auth.py [hunk 1]
```diff
@@ -139,7 +139,10 @@ def requests_auth(username, password, **kwargs):
 class ApiTokenAuth:
     @staticmethod
     def requests_auth(username, password, **kwargs):
-        if StaxConfig.expiration and StaxConfig.expiration > datetime.now(timezone.utc):
+        # Minimize the potentical for token to expire while still being used for auth (say within a lambda function)
+        if StaxConfig.expiration and StaxConfig.expiration - timedelta(
+            minutes=10
```
Comment (wolfeidau) [inline staxapp/auth.py:144]: I think 1 minute would be enough.

Comment (wolfeidau) [reply staxapp/auth.py:144]: Would be good to add this line to RootAuth section as well, just above it!

**Review by wolfeidau — CHANGES_REQUESTED** [review]
Just some small things.

File: staxapp/auth.py (hunk 1 above)
Comment (richardkeit) [reply staxapp/auth.py:144]: @wolfeidau - where is `RootAuth` used? I don't see any references for it (except testing)

**Comment by Anton0** [comment]
@richardkeit you will likely need to rebase or merge from master to get some updates to the codecov github action, apologies

File: staxapp/auth.py (hunk 1 above)
Comment (wolfeidau) [reply staxapp/auth.py]: @richardkeit it is a legacy thing, I need to check if we can remove it.

**Review by wolfeidau — APPROVED** [review]

**Pull request merged**